tools/gn

```diff
@@ -212,19 +238,39 @@ def to_gn_args(args):
       # The GN arg is not available in the windows toolchain.
       gn_args['enable_lto'] = enable_lto
 
-    if args.target_os:
+    # Set OS, CPU arch for host or target build.
+    if is_host_build(args):
+      gn_args['host_os'] = get_host_os()
+      gn_args['host_cpu'] = get_host_cpu(args)
+      gn_args['target_os'] = gn_args['host_os']
+      gn_args['target_cpu'] = get_target_cpu(args)
+      gn_args['dart_target_arch'] = gn_args['target_cpu']
+    else:
       gn_args['target_os'] = args.target_os
-    gn_args['target_cpu'] = get_target_cpu(args)
-    gn_args['dart_target_arch'] = gn_args['target_cpu']
+      gn_args['target_cpu'] = get_target_cpu(args)
+      gn_args['dart_target_arch'] = gn_args['target_cpu']
```

Comment:
Setting target_cpu and dart_target_arch here could be extracted out and made unconditional, but I'm not sure it makes the code any simpler to reason about. Maybe worth it with dart_target_arch...